webview_panels/src/modules/queryPanel/components/perspective/PerspectivePlugins.js

```diff
@@ -0,0 +1,174 @@
+
+    // Styling the image element
+    img.style.paddingTop = "5px";
+
+    // Styling the span element
+    span.style.marginLeft = "5px";
+    span.style.color = "var(--primary-color)";
+
+    // Styling the div element
+    div.style.display = "flex";
+    div.style.alignItems = "center";
+    div.style.justifyContent = "space-between";
+
+    // Styling the text element
+    text.style.overflow = "hidden";
+    text.style.textOverflow = "ellipsis";
+    text.style.whiteSpace = "nowrap";
+    text.style.width = "100px";
+
+    // Appending the image to the span element and the text to the div element
+    // Appending the div element to the td element
+    span.appendChild(img);
+    div.appendChild(text);
+    div.appendChild(span);
+    td.appendChild(div);
+  }
+}
+
+// Adds click event to the td element
+function makeClickEvent(td, metadata) {
+  // If string length is greater than 20, truncate and add ellipsis
+  if (metadata.value.length > 20) {
+    // td.textContent = metadata.value.slice(0, 20) + "...";
+    td.style.cursor = "pointer";
+  }
+
+  const columnName = metadata.column_header;
+  if (isJson(metadata.value)) {
+    appendImage(td);
+    td.style.cursor = "pointer";
+    // If string is a valid JSON, add click event to view JSON
+    td.onclick = () => {
+      dispatchCustomEvent(
+        "string-json-viewer",
+        metadata.value,
+        columnName,
+        "json"
+      );
+    };
+  } else if (metadata.value.length > 20) {
+    appendImage(td);
+    // else if string is greater than 20, add click event to view full string
+    td.onclick = () => {
+      dispatchCustomEvent(
+        "string-json-viewer",
+        metadata.value,
+        columnName,
+        "string"
+      );
+    };
+  }
+}
+
+// Custom perspective plugin to add click event to the td element
+class CustomDatagridPlugin extends customElements.get(
+  "perspective-viewer-datagrid"
+) {
+  get name() {
+    return "Custom Datagrid";
+  }
+
+  async styleListener() {
+    const viewer = this.parentElement;
+    const datagrid = this.regular_table;
+    if (this._dirty) {
+      await this.refresh_cache();
+    }
+
+    for (const td of datagrid.querySelectorAll("td")) {
+      const metadata = datagrid.getMeta(td);
+
+      let type;
```

Comment:
Better to avoid mutable variables. You could extract this logic to separate function and return the type